aws/cloudwatch-alarms-delivery/main.tf

```diff
@@ -0,0 +1,126 @@
+locals {
+  function_name = "${var.name}-cloudwatch-alarms-delivery-${random_id.unique_id.dec}"
+}
+
+resource "aws_sns_topic_subscription" "cloudwatch_opsgenie_delivery" {
+  count = var.endpoint == "Opsgenie" ? 1 : 0
```

Comment:
What do you think about splitting modules by both source and destination? Looking through the resources, it looks like every resource is used by either Opsgenie or Sentry, but none of them are used by both. This would also make it easier to avoid doing things like making a Lambda role that won't be used. Then you'd have `aws/cloudwatch-alarms-delivery/opsgenie` and `aws/cloudwatch-alarms-delivery-sentry`.